### PR TITLE
add: wrapper ensures exit code 1 is reserved for findings

### PR DESCRIPTION
**Pull request by brendongo**
PR checklist:

- [ ] Documentation is up-to-date
- [ ] Changelog is up-to-date
- [ ] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

**Comment by brendongo** [comment]
~~TODO: add more comprehensive tests~~ Done

File: semgrep/semgrep/commands/wrapper.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        # than warning are handled twice
+        logger = getLogger("semgrep")
+        logger.propagate = False
+        metric_manager.set_version(__VERSION__)
+
+        try:
+            func(*args, **kwargs)
+        # Catch custom exception, output the right message and exit
+        except SemgrepError as e:
+            metric_manager.set_return_code(e.code)
```
Comment (underyx) [inline semgrep/semgrep/commands/wrapper.py:40]:
```suggestion
            metric_manager.set_return_code(FATAL_EXIT_CODE)
            sys.exit(FATAL_EXIT_CODE)
```
how come we don't do this here?

**Review by underyx — APPROVED** [review]

File: semgrep/semgrep/commands/wrapper.py (hunk 1 above)
Comment (brendongo) [reply semgrep/semgrep/commands/wrapper.py:40]: Cause the code could be more specific than just exit code 2

File: semgrep/semgrep/commands/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            sys.exit(e.code)
+        except Exception as e:
+            logger.exception(e)
+            sys.exit(FATAL_EXIT_CODE)
```
Comment (underyx) [inline semgrep/semgrep/commands/wrapper.py:45]:
```suggestion
            metric_manager.set_return_code(FATAL_EXIT_CODE)
            sys.exit(FATAL_EXIT_CODE)
```
how come we don't do this here?

**Pull request merged**